skpro/utils/set_output.py

```diff
@@ -0,0 +1,157 @@
+"""Utilities for set_output functionality."""
+
+__author__ = ["julian-fong"]
+
+import warnings
+from copy import deepcopy
+
+import pandas as pd
+
+from skpro.utils.validation._dependencies import _check_soft_dependencies
+
+SUPPORTED_OUTPUTS = ["pandas"]
+
+if _check_soft_dependencies(["polars", "pyarrow"], severity="none"):
+    SUPPORTED_OUTPUTS.append("polars")
+
+
+def check_n_level_of_dataframe(X_input, index_pandas=False):
+    """Convert function to check the number of levels inside a pandas Frame.
+
+    Parameters
+    ----------
+    X_input : polars or pandas DataFrame
+        A given polars or pandas DataFrame. Note that the polars portion of this
+        code requires the soft dependencies polars and pyarrow to be installed
+
+    index_pandas : bool
```

Comment:
strange choice of parameter, unintuitive. I would say, change either to string (`"index"`, `"columns"`), or require the user to pass an index in the first place.

Reply:
perhaps we rename is to `axis` and specify 0 or 1?